src/Microsoft.VisualStudio.Threading.Shared/JoinableTask.cs

```diff
@@ -465,7 +465,8 @@ private bool SynchronouslyBlockingThreadPool
             get
             {
                 return (this.state & JoinableTaskFlags.StartedSynchronously) == JoinableTaskFlags.StartedSynchronously
-                    && (this.state & JoinableTaskFlags.StartedOnMainThread) == JoinableTaskFlags.None;
+                    && (this.state & JoinableTaskFlags.StartedOnMainThread) == JoinableTaskFlags.None
+                    && (this.state & JoinableTaskFlags.CompleteRequested) != JoinableTaskFlags.CompleteRequested;
```

Comment:
It would be good if this is consistent, ie either use "== None" or "!= flag" on both conditions.